### PR TITLE
#RI-4989 - refactor json type

### PR DESCRIPTION
**Pull request by romansergeenkosofteq**

### CODE REVIEW

**Review by AmirAllayarovSofteq — APPROVED** [review]
Looks really good

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/RejsonDetailsWrapper.tsx [hunk 1]
```diff
@@ -73,35 +82,32 @@ const RejsonDetailsWrapper = (props: Props) => {
...
-                  shouldRejsonDataBeDownloaded={!downloaded}
-                  handleSubmitJsonUpdateValue={handleSubmitJsonUpdateValue}
-                  handleSubmitUpdateValue={handleEditValueUpdate}
-                />
-              )}
-            </div>
+        <div className="flex-column" style={{ flex: '1', height: '100%' }}>
+          <div
+            data-testid="json-details"
+            className={`${[styles.container].join(' ')}`}
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/RejsonDetailsWrapper.tsx:88]: let it be `className={styles.container}`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/add-item/AddItem.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  const [key, setKey] = useState<string>('')
+  const [value, setValue] = useState<string>('')
+  const [error, setError] = useState<Nullable<string>>(null)
+
+  useEffect(() => {
+    setError(null)
+  }, [key, value])
+
+  const handleOnEsc = (e: KeyboardEvent) => {
+    if (e.code?.toLowerCase() === 'escape') {
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/add-item/AddItem.tsx:44]: we have enum keys, `keys.ESCAPE`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-entire-item-action/EditEntireItemAction.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  const {
+    initialValue,
+    onCancel,
+    onSubmit
+  } = props
+  const [value, setValue] = useState<string>(initialValue)
+  const [error, setError] = useState<Nullable<string>>(null)
+
+  const handleOnEsc = (e: KeyboardEvent) => {
+    if (e.code?.toLowerCase() === 'escape') {
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-entire-item-action/EditEntireItemAction.tsx:35]: the same `keys.ESCAPE`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-entire-item-action/EditEntireItemAction.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      e.stopPropagation()
+      onCancel?.()
+    }
+  }
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    const error: null | string = validateRejsonValue(value)
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-entire-item-action/EditEntireItemAction.tsx:44]: `Nullable<string>`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-item-field-action/EditItemFieldAction.tsx [hunk 5]
```diff
@@ -40,21 +32,21 @@ const EditItemFieldAction = ({
...
       <PopoverDelete
         header={createDeleteFieldHeader(keyName)}
         text={createDeleteFieldMessage(bufferToString(selectedKey))}
         item={keyName}
-        suffix={type}
         deleting={deleting}
         closePopover={() => setDeleting('')}
         updateLoading={false}
-        showPopover={(item) => { setDeleting(`${item}${type}`) }}
+        showPopover={(item) => { setDeleting(item) }}
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/components/edit-item-field-action/EditItemFieldAction.tsx:44]: `showPopover={setDeliting}`

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-details/RejsonDetails.spec.tsx [hunk 6]
```diff
@@ -161,26 +169,24 @@ describe('RejsonDetails', () => {
...
+      isDownloaded
     />)
 
     fireEvent.click(screen.getByTestId('add-object-btn'))
     expect(screen.getByTestId('json-key')).toBeInTheDocument()
     expect(screen.getByTestId('json-value')).toBeInTheDocument()
   })
 
-  it('should be able to add proper key value into json object', () => {
+  it.skip('should be able to add proper key value into json object', () => {
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-details/RejsonDetails.spec.tsx:181]: Is it need to skip all these tests?

File: redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-object/RejsonObject.tsx [hunk 7]
```diff
@@ -1,209 +1,108 @@
...
 
-    const body: IJSONObject = {
-      previous_path: path,
-      new_key: newKey,
-      path: `${path}[${newKey}]`,
-      operation: 'update',
-      value: newValue
+    try {
+      const unescapedKey = JSON.parse(key as string)
+      const updatedPath = unescapedKey.includes('"') ? `${path}['${unescapedKey}']` : `${path}["${unescapedKey}"]`
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-object/RejsonObject.tsx:68]: you have almost the same logic in rejesonDetails, let move this logic to some util, like `getUnescapedKey(key)`

**Review by zalenskiSofteq — APPROVED** [review]

**Pull request merged**